backend/__init__.py

```diff
@@ -273,9 +274,10 @@ def handle_invalid_usage(error):
 def handle_insufficient_permission(error):
     """ Handler for insufficient permission to access a method.
         This is not the error handler for insufficient permission to update
-        a project or user.
+        a project or user. It delivers a 404 as users should not see endpoints,
+        where they don't have permissions to.
```

Comment:
this handler is for cases where the user knows the endpoint but doesn't have permission to use it.
the other errorhandler is for cases where the user should not be aware of the endpoint